### PR TITLE
Bug buffer overrun in debug macro

### PR DESCRIPTION
**Pull request by dbenhur**
@axos88 @jfoy @kiyose Please review.  mobile-search-gateway encountered this buffer overrun over the weekend, caused lots of log spam and out-of-disk alarms by enabling debug_level.

I added accessors for curdepth and maxdepth so I can instrument some stuff in Ruby to understand what drives such deep tracing of function entry/exit.

### CODE REVIEW

File: lib/rb/ext/debug.h [hunk 1]
```diff
@@ -24,12 +25,10 @@ extern int curdepth;
 
 #define DEBUGS(s) DEBUGF("%s%s\n", separator, s)
```
Comment (jfoy) [inline lib/rb/ext/debug.h]: Stray `separator` token.

**Comment by jfoy** [comment]
Otherwise LGTM

File: lib/rb/ext/debug.h [hunk 2]
```diff
@@ -13,7 +14,7 @@ extern int curdepth;
 #define LVL_LOG 4
 #define LVL_DEBUG 5
 
-#define DEBUG_MSG(lvl, fmt, ...) ((debug_level >= lvl) ? printf("%s", separator), printf(fmt, ##__VA_ARGS__), printf("\n") : 0)
+#define DEBUG_MSG(lvl, fmt, ...) ((debug_level >= lvl) ? printf("%*s", curdepth, ""), printf(fmt, ##__VA_ARGS__), printf("\n") : 0)
```
Comment (axos88) [inline lib/rb/ext/debug.h]: Nice trick

**Comment by axos88** [comment]
LGTM

**Pull request merged**